src/RenderCapableTrait.php

```diff
@@ -2,23 +2,44 @@
 
 namespace Dhii\Output;
 
+use Dhii\Output\Exception\TemplateRenderExceptionInterface;
 use Exception as RootException;
 use Dhii\Output\Exception\CouldNotRenderExceptionInterface;
 use Dhii\Output\Exception\RendererExceptionInterface;
 use Psr\Container\ContainerInterface;
 use Dhii\Util\String\StringableInterface as Stringable;
 
 /**
- * Common functionality for template rendering.
+ * Functionality for rendering a template.
  *
- * @since 0.1
+ * @since [*next-version*]
  */
-abstract class AbstractTemplateBlock extends AbstractBlock
+trait RenderCapableTrait
```

Comment:
This name is a bit misleading as it implies "functionality for rendering" in general. Consider `TemplateRenderCapableTrait`.

**Edit**: After carefully reading the code in this trait, it seems like this is intended for something that is already aware of the template and context, like a block that internally uses a template for rendering. If this _is_ the case, consider `TemplateBlockRenderCapableTrait`.